features/networking/multus.feature

```diff
@@ -1394,4 +1394,61 @@ Feature: Multus-CNI related scenarios
       | <%= project.name %>:macvlan-bridge-whereabouts-pod1:<%= cb.pod_uid %> |
       | <%= cb.pod_eth0_ip %>                                                 |
       | <%= cb.pod_net1_ip %>                                                 |
+      
+  # @author weliang@redhat.com
+  # @case_id OCP-31999
+  @admin
+  Scenario: Whereabouts with exclude IP address	
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    Given I obtain test data file "networking/multus-cni/NetworkAttachmentDefinitions/whereabouts-excludeIP.yaml"
+    When I run oc create as admin over "whereabouts-excludeIP.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %> |
+    Then the step should succeed
+    
+    # Create a pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | macvlan-bridge-whereabouts-pod1 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip           |
+      | ["spec"]["containers"][0]["name"]                          | whereabouts-excludeip           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod1" becomes ready
+    # Check the created pod has correct ip 
+    When I execute on the "macvlan-bridge-whereabouts-pod1" pod:
+      | bash | -c | /usr/sbin/ip -4 -brief a |
+    Then the output should contain:
+      | 192.168.10.4 |
+
+    # Create second pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | macvlan-bridge-whereabouts-pod2 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip           |
+      | ["spec"]["containers"][0]["name"]                          | whereabouts-excludeip           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod2" becomes ready
```

Comment:
Instead of `create pod1, wait pod1 ready; then create pod2, wait pod2 ready`, we can `create pod1, create pod2, wait pod1 ready, wait pod2 ready`.

E.g, let's assume create pod takes 1 second, and wait pod ready takes 30 second.
`create pod1, wait pod1 ready; then create pod2, wait pod2 ready` need about 62 (1+30+1+30) seconds.
While `create pod1, create pod2, wait pod1 ready, wait pod2 ready` need about 32 (1+1+30) seconds.

Reply:
@liangxia Agree, your suggestion is good for other test case. But not for this case. The purpose of this test case is make sure pod1 get  ip address of x.x.x.x and pod2 to get ip address of y.y.y.y. From manual testing, when creating pod1 and pod2 at same time, pod1 not always to get x.x.x.x, this is OK for manual testing, but it will cause automation checking fail .Thanks!

Reply:
Ack